articles/forward-and-backward-chaining-in-ai/index.md

```diff
@@ -0,0 +1,119 @@
+
+In this type of chaining, the inference engine starts by evaluating existing facts, derivations, and conditions before deducing new information. An endpoint (goal) is achieved through the manipulation of knowledge that exists in the knowledge base. 
+
+![Forward Chaining](/engineering-education/forward-and-backward-chaining-in-ai/forward-chaining.jpg)
+
+[Image Source: Tutorials Point](https://www.tutorialspoint.com/artificial_intelligence/images/forward_chaining.jpg)
+
+Forward chaining can be used in planning, monitoring, control, and interpretation applications.
+
+#### Properties of Forward Chaining
+* The process uses a down-up approach (bottom to top).
+* It starts from an initial state and uses facts to make a conclusion.
+* This approach is data-driven.
+* It is employed in expert systems and production rule system.
+
+#### Example of Forward Chaining
+A simple example of forward chaining can be explained in the following sequence.
+
+A
+
+A->B
+
+B
+
+A is the starting point. A->B represents a fact. This fact is used to achieve a decision B. 
+
+A practical example will go as follows;
+
+Tom is running (A)
+
+If a person is running, he will sweat (A->B)
+
+Therefore, Tom is sweating. (B)
+
+#### Advantages
+* It can be used to draw multiple conclusions.
+* It provides a good basis for arriving at conclusions.
+* It is more flexible than backward chaining because it does not have a limitation on the data derived from it.
+
+#### Disadvantages
+* The process of forward chaining may be time-consuming. It may take a lot of time to eliminate and synchronize available data.
+* Unlike backward chaining, the hypothesis for this type of chaining is not very clear. The former uses a goal-driven method that arrives at conclusions efficiently. 
+
+### Backward Chaining
+Backward chaining is a concept in artificial intelligence that involves backtracking from the endpoint or goal to steps that led to the endpoint. This type of chaining starts from the goal and moves backward to comprehend the steps that were taken to attain this goal. 
+
+The backtracking process can also enable a person to establish logical steps that can be used to find other important solutions.
+
+![Backward Chaining](/engineering-education/forward-and-backward-chaining-in-ai/backward-chaining.jpg)
+
+[Image Source: Quora](https://qph.fs.quoracdn.net/main-qimg-72fa83318b4bf6238b5a0da493a56277)
+
+Backward chaining can be used in debugging, diagnostics, and prescription applications.
+
+#### Properties of Backward Chaining
+* The process uses an up-down approach (top to bottom).
+* It is a goal-driven method of reasoning.
+* The endpoint (goal) is subdivided into sub-goals to prove the truth of facts.
+* Backward chaining algorithm is employed in inference engines, game theories, and complex database systems.
+* The modus ponens inference rule is used as the basis for the backward chaining process. 

```

Comment:
Mention what modus ponens is and how it is applicable in backward chaining.

Reply:
Done.